pkg/snapstore/swift_snapstore.go

```diff
@@ -181,50 +233,74 @@ func readSwiftCredentialFiles(dirname string) (*clientconfig.ClientOpts, error)
 	}, nil
 }
 
-func readSwiftCredentialDir(dirname string) (*swiftCredentials, error) {
+func readSwiftCredentialDir(dirName string) (*swiftCredentials, error) {
 	cred := &swiftCredentials{}
-	files, err := os.ReadDir(dirname)
+	files, err := os.ReadDir(dirName)
 	if err != nil {
 		return nil, err
 	}
 
+	if cred.AuthType, err = getSwiftCredentialAuthType(files); err != nil {
```

Comment:
Is there a case where we have Username/Password file and ApplicationCredentialID/ApplicationCredentialName/ApplicationCredentialSecret files are both present? 
If that is the case then the flow might be a bit indeterministic

Reply:
>  Username/Password file and ApplicationCredentialID/ApplicationCredentialName/ApplicationCredentialSecret files are both present?

I'm assuming that shouldn't be the case as I have also mentioned in the [doc examples of secrets](https://github.com/gardener/etcd-backup-restore/pull/580/files#diff-ee65c41e7448f5d9a9cb70e797bf558bb21d2bf277348daf329cf8e6261084eaR39-R52).

Reply:
@aaronfern is right. You should not assume and have a validation which prevents this from happening.